[BUG] BOARD_BTT_SKR_V1_4 "SD_DETECT_PIN" redefined

```
In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/../pins/pins.h:375,
                 from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:32,
                 from Marlin/src/HAL/HAL_LPC1768/watchdog.cpp:25:
Marlin/src/HAL/HAL_LPC1768/../../inc/../pins/lpc1768/pins_BTT_SKR_V1_4.h:267: warning: "SD_DETECT_PIN" redefined
  267 |     #define SD_DETECT_PIN  P1_31   // (49) (NOT 5V tolerant)
      | 
In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/../pins/lpc1768/pins_BTT_SKR_V1_4.h:156,
                 from Marlin/src/HAL/HAL_LPC1768/../../inc/../pins/pins.h:375,
                 from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:32,
                 from Marlin/src/HAL/HAL_LPC1768/watchdog.cpp:25:
Marlin/src/HAL/HAL_LPC1768/../../inc/../pins/lpc1768/pins_BTT_SKR_common.h:105: note: this is the location of the previous definition
  105 |   #define SD_DETECT_PIN    P0_27
```

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -264,7 +264,11 @@
     #define LCD_PINS_D4    P1_20
 
     #define LCD_SDSS       P0_16   // (16) J3-7 & AUX-4
-    #define SD_DETECT_PIN  P1_31   // (49) (NOT 5V tolerant)
+
+    #if SD_CONNECTION_IS(LCD)
+      #undef SD_DETECT_PIN
+      #define SD_DETECT_PIN  P1_31   // (49) (NOT 5V tolerant)
+    #endif
 
     #if ENABLED(FYSETC_MINI_12864)
       #define DOGLCD_CS    P1_18

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -151,11 +151,6 @@
 #define TEMP_BED_PIN       P0_25_A2   // A0 (T0) - (67) - TEMP_BED_PIN
 
 //
-// Include common SKR pins
-//
-#include "pins_BTT_SKR_common.h"
-
-//
 // Software SPI pins for TMC2130 stepper drivers
 //
 #if ENABLED(TMC_USE_SW_SPI)
@@ -332,3 +327,8 @@
  *   P0_27  (57) (Open collector)
  *   P0_28  (58) (Open collector)
  */
+
+//
+// Include common SKR pins
+//
+#include "pins_BTT_SKR_common.h"

--- a/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
+++ b/Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_4.h
@@ -266,8 +266,7 @@
     #define LCD_SDSS       P0_16   // (16) J3-7 & AUX-4
 
     #if SD_CONNECTION_IS(LCD)
-      #undef SD_DETECT_PIN
-      #define SD_DETECT_PIN  P1_31   // (49) (NOT 5V tolerant)
+      #define SD_DETECT_PIN P1_31  // (49) (NOT 5V tolerant)
     #endif
 
     #if ENABLED(FYSETC_MINI_12864)

